### ISSUE TITLE
Raster fails when using DOM image as source

### ISSUE THREAD

**Issue by arel** [paperjs/paper.js#1645]
# Description/Steps to reproduce

I am experiencing an error creating a new `Raster` object when using a DOM image as the source.
I've noticed that instantiating a `Raster` in different ways produces different results however:

```javascript
const domImg = new window.Image(100, 100);
domImg.src = 'http://paperjs.org/about/paper-js.gif';

const ok = new paper.Raster(domImg);
const fails = new paper.Raster({source: domImg});
```

If I define an `onError(error)` function on the latter case, I get:
```javascript
JSON.stringify(error) // "{"event":{"isTrusted":true},"type":"error"}"
error.event.target   // <img src=​"[object HTMLImageElement]​">​
```

# Link to reproduction test-case

See example above.

# Expected result

Image should load in both cases from DOM as specified in the documentation.

# Additional information

paper 0.12.0, node v8.12.0, Ubuntu 16.04.5 LTS

**Comment by sasensi** [comment]
Hi, thank you for the report.

As a workaround, waiting for a correction, you can use [raster.image](http://paperjs.org/reference/raster/#image) property instead.
See this [sketch](http://sketch.paperjs.org/#V/0.11.8/S/fZLBbsIwDIZfxeoF0KBh5daJ2y4ctknbETiE4pWsa1IlLtWEePc5CWVijOVgKc5v/1+cHBIta0zy5K1CKnbJOCnM1u+FWNmVFgIeX54AP7FGTVAYTVJppUtQtSwxSnzcSwtbUy/qEuagsYOFPx/eT6dj4DB68KIoSJ0tWDTYETW5EI1s0H641NhSyI1pKWYmnCrV+yAU/vgw0DN3t9IRWugU7QJgoAHpQNqyDajDztjKjS4JTXWiew31wwgU4UyVNsYpUkazaK+wSwtuxC53sJyEi/i47sWOh1IQbllMtsVrzoV2JDUpSQitO88MGmv4fvSV/mbLLuEO/gR4hbL8NF6fPPbE2T/IV8TZTeRoP7u07z1macSenwDO6dvW8c3XZ+VfvvzTNhZl1RilySX5cn38Bg==) for a demonstration.

```javascript
new Raster({
    image: domImg
});

var raster= new Raster();
raster.image = domImg;
```

About what we should do, I see 2 possibilities:
- Changing documentation to make it describe actual behavior: only a URL can be set as the source.
This seems problematic to me because the getter can return either an image, a canvas or a URL and there would be an inconsistency between getter and setter.
- Make source setter support either image, canvas or URL as described in the documentation.
I think this makes more sense as it was the legacy behavior as we can see browsing the source history.
So the user might expect that.

@lehni, what do you think ?

**Comment by lehni** [comment]
Yes, the documentation is wrong here. `Raster#source` is only expected to be a string, and can either be a URL or a data URL. `Raster#image` accepts a DOM image or a canvas object. I'll change the documentation.

**Comment by lehni** [comment]
BTW, the constructor itself supports both versions: `new Raster(domImage)` and `new Raster(url)`
